app/src/parser_impl.c

```diff
@@ -0,0 +1,873 @@
+    CHECK_ERROR(_readMapSize(c, &paramsSize))
+
+    CHECK_ERROR(_findKey(c, KEY_APARAMS_MANAGER))
+    CHECK_ERROR(_readBinFixed(c, asset_config->params.manager, sizeof(asset_config->params.manager)))
+    addItem(1);
+
+    CHECK_ERROR(_findKey(c, KEY_APARAMS_RESERVE))
+    CHECK_ERROR(_readBinFixed(c, asset_config->params.reserve, sizeof(asset_config->params.reserve)))
+    addItem(2);
+
+    CHECK_ERROR(_findKey(c, KEY_APARAMS_FREEZE))
+    CHECK_ERROR(_readBinFixed(c, asset_config->params.freeze, sizeof(asset_config->params.freeze)))
+    addItem(3);
+
+    CHECK_ERROR(_findKey(c, KEY_APARAMS_CLAWBACK))
+    CHECK_ERROR(_readBinFixed(c, asset_config->params.clawback, sizeof(asset_config->params.clawback)))
+    addItem(4);
+
+    tx_num_items += 4;
+
+    // These items won't be displayed
+    if (_findKey(c, KEY_APARAMS_TOTAL) == parser_ok) {
+        CHECK_ERROR(_readInteger(c, &asset_config->params.total))
+    }
+    if (_findKey(c, KEY_APARAMS_DECIMALS) == parser_ok) {
+        CHECK_ERROR(_readInteger(c, &asset_config->params.decimals))
+    }
+    if (_findKey(c, KEY_APARAMS_DEF_FROZEN) == parser_ok) {
+        CHECK_ERROR(_readBool(c, &asset_config->params.default_frozen))
+    }
+    if (_findKey(c, KEY_APARAMS_UNIT_NAME) == parser_ok) {
+        CHECK_ERROR(_readString(c, (uint8_t*)asset_config->params.unitname, sizeof(asset_config->params.unitname)))
+    }
+    if (_findKey(c, KEY_APARAMS_ASSET_NAME) == parser_ok) {
+        CHECK_ERROR(_readString(c, (uint8_t*)asset_config->params.assetname, sizeof(asset_config->params.assetname)))
+    }
+    if (_findKey(c, KEY_APARAMS_URL) == parser_ok) {
+        CHECK_ERROR(_readString(c, (uint8_t*)asset_config->params.url, sizeof(asset_config->params.url)))
+    }
+    if (_findKey(c, KEY_APARAMS_METADATA_HASH) == parser_ok) {
+        CHECK_ERROR(_readBinFixed(c, asset_config->params.metadata_hash, sizeof(asset_config->params.metadata_hash)))
+    }
+
+    return parser_ok;
+}
+
+parser_error_t _readAppArgs(parser_context_t *c, uint8_t args[][MAX_ARGLEN], size_t args_len[], size_t *argsSize, size_t maxArgs)
+{
+    CHECK_ERROR(_readArraySize(c, argsSize))
+
+    if (*argsSize > maxArgs) {
+        return parser_msgpack_array_too_big;
+    }
+
+    for (size_t i = 0; i < *argsSize; i++) {
+        CHECK_ERROR(_readBin(c, args[i], (uint16_t*)&args_len[i], MAX_ARGLEN))
+    }
+
+    return parser_ok;
+}
+
+parser_error_t _readAccounts(parser_context_t *c, uint8_t accounts[][32], size_t *numAccounts, size_t maxAccounts)
+{
+    CHECK_ERROR(_readArraySize(c, numAccounts))
+
+    if (*numAccounts > maxAccounts) {
+        return parser_msgpack_array_too_big;
+    }
+
+    for (size_t i = 0; i < *numAccounts; i++) {
+        CHECK_ERROR(_readBinFixed(c, accounts[i], sizeof(accounts[0])))
+    }
+
+    return parser_ok;
+}
+
+parser_error_t _readStateSchema(parser_context_t *c, state_schema *schema)
+{
+    size_t mapSize = 0;
+    CHECK_ERROR(_readMapSize(c, &mapSize))
+    uint8_t key[32];
```

Comment:
is `32` always enough?
assuming that is a string, do we assume zero termination or it should be allocating `33` and doing a memzero before to ensure termination?